### ISSUE TITLE
PHP: Error decoding with missing field (old message definition)

### ISSUE THREAD

**Issue by NicklasWallgren** [google/protobuf#2552]
It isn't possible to decode/parse using an old message definition. If a field is missing in the message definition the "parseFieldFromStream" will fail with a fatal error. 

"You can add new fields to your message formats without breaking backwards-compatibility; old binaries simply ignore the new field when parsing."

```
parseFromStream()
$field = $this->desc->getFieldByNumber($number); // $field is null
...
parseFieldFromStream()
$field_type = $field->getType(); // fatal error
```

A dirty quick-fix is to check for null fields in the parseFromStream() method, and continue if encountered.

```
$number = GPBWire::getTagFieldNumber($tag);
$field = $this->desc->getFieldByNumber($number);

// Check whether we retrieved a known field
if ($field == null) {
    continue;
}

if (!$this->parseFieldFromStream($tag, $input, $field)) {
    return false;
}
```

**Comment by NicklasWallgren** [comment]
https://github.com/NicklasWallgren/protobuf/commit/4f29e1337f40111494d2131dc613a3ab6ea94b8f

**Comment by TeBoring** [comment]
@NicklasWallgren Thanks, could you send this pull request to 3.2.x branch and also add a test?

**Comment by TeBoring** [comment]
Fixed in 60d4f8af483e0234ee5b7f7544bcd578a34412ad
